docs/contributing.md

```diff
@@ -0,0 +1,71 @@
+# Contributing definitions
+
+Contributing library definitions is as easy as sending a pull request!
+
+All definitions sit under the
+[/definitions](https://github.com/flowtype/flow-typed/tree/master/definitions)
+directory. They all must follow the following directory structure and naming
+format:
+
+```
+└ definitions/npm/
+  ├ yargs_v4.x.x/           # <-- The name of the library, followed by _v<VERSION>
+  | |
+  | ├ flow_v0.23.x/         # <-- A folder containing libdefs tested against the
+  | | |                     #     specified version(s) of Flow (v0.23.x in this
+  | | |                     #     case).
+  | | |
+  | | └ yargs_v4.x.x.js     # <-- The libdef file meant for the Flow version
+  | |                       #     specified by the containing directory's name.
+  | |                       #     Must be named `<LIB>_v<VERSION>.js`.
+  | |
+  | ├ flow_v0.19.x-v0.22.x/ # <-- A folder containing libdefs tested against a
+  | | |                     #     different range of Flow versions:
+  | | |                     #     Anything from v0.19.x to v0.22.x (inclusive)
+  | | |
+  | | ├ yargs_v4.x.x.js     # <-- The libdef file for versions of Flow from
+  | | |                     #     v0.19.x to v0.22.x (inclusive)
+  | | |
+  | | └ test_yargs.js       # <-- Tests in this directory only apply to the
+  | |                       #     adjacent libdef (and thus, are specific to
+  | |                       #     the libdefs for this specific Flow version)
+  | |
+  | └ test_yargs.js         # <-- Tests in this directory apply to libdefs for
+  |                         #     all versions of Flow.
+  ├ color_v0.7.x/
+  ├ ...
+```
+
+Versions are semver versions with some restrictions:
+
+* All of MAJOR, MINOR, and PATCH versions must be specified. It's acceptable to
+  specify `x` in place of a number for MINOR and PATCH, but MAJOR cannot be `x`.
+* Library versions cannot specify a semver range, but Flow versions can of the
+  following forms:
+  * **`flow_v0.22.x-`**: Flow v0.22.x and above
+  * **`flow_v0.22.x-v0.28.x`**: Flow versions v0.22.x up to v0.28.x (inclusive)
+  * **`flow_-v0.22.x`**: Every version under (and including) Flow v0.22.x
+
+**We structure files this way to enable automated testing and tooling.**
+Tests ensure that library definitions continue to work as expected and the
+`flow-typed` tooling ensures that it's as easy as possible to find and install
+library definitions.
+
+When submitting new library definitions, opt for `flow_v0.83.x-` instead of `flow_all`. Starting with Flow v0.83 you can use the syntax `declare module.exports: type;`.
```

Comment:
This is the first mention that `flow_all` is possible do you want to add that to the above list but mark it as not recommended?

Reply:
Honestly we don't even need that line, most contributors don't use flow_all anyways. I'll just remove the entire thing.